DEHPCommon/CommonUserInterface/ViewModels/Common/LoginViewModel.cs

```diff
@@ -0,0 +1,432 @@
+    using System.Linq;
+    using System.Reactive;
+    using System.Threading.Tasks;
+
+    using CDP4Common.SiteDirectoryData;
+
+    using CDP4Dal;
+    using CDP4Dal.DAL;
+
+    using CDP4ServicesDal;
+
+    using CDP4WspDal;
+
+    using DEHPCommon.CommonUserInterface.ViewModels.Rows;
+
+    using Microsoft.Win32;
+
+    using ReactiveUI;
+
+    /// <summary>
+    /// The view-model for the Login that allows users to connect to different datasources
+    /// </summary>
+    public class LoginViewModel : ReactiveObject
+    {
+        /// <summary>
+        /// Gets or sets datasource server type
+        /// </summary>
+        public static KeyValuePair<string, string>[] DataSourceList { get; } = {
+            new KeyValuePair<string, string>("CDP", "CDP4 WebServices"),
+            new KeyValuePair<string, string>("OCDT", "OCDT WSP Server"),
+        };
+
+        /// <summary>
+        /// Backing field for the <see cref="ServerType"/> property
+        /// </summary>
+        private KeyValuePair<string, string> serverType;
+
+        /// <summary>
+        /// Gets or sets server serverType value
+        /// </summary>
+        public KeyValuePair<string, string> ServerType
+        {
+            get => this.serverType;
+
+            set => this.RaiseAndSetIfChanged(ref this.serverType, value);
+        }
+
+        /// <summary>
+        /// Backing field for the <see cref="Username"/> property
+        /// </summary>
+        private string username;
+
+        /// <summary>
+        /// Gets or sets server username value
+        /// </summary>
+        public string UserName
+        {
+            get => this.username;
+
+            set => this.RaiseAndSetIfChanged(ref this.username, value);
+        }
+
+        /// <summary>
+        /// Backing field for the <see cref="Password"/> property
+        /// </summary>
+        private string password;
+
+        /// <summary>
+        /// Gets or sets server password value
+        /// </summary>
+        public string Password
+        {
+            get => this.password;
+
+            set => this.RaiseAndSetIfChanged(ref this.password, value);
+        }
+
+        /// <summary>
+        /// Backing field for the <see cref="Uri"/> property
+        /// </summary>
+        private string uri;
+
+        /// <summary>
+        /// Gets or sets server uri
+        /// </summary>
+        public string Uri
+        {
+            get => this.uri;
+
+            set => this.RaiseAndSetIfChanged(ref this.uri, value);
+        }
+
+        /// <summary>
+        /// The backing field for available data access layer <see cref="IDal"/>
+        /// </summary>
+        private IDal dal;
+
+        /// <summary>
+        /// Backing field for the <see cref="ISession"/> property
+        /// </summary>
+        private ISession session;
+
+        /// <summary>
+        /// Gets or sets login succesfully flag
+        /// </summary>
+        public ISession ServerSession
```

Comment:
The session in this control is not needed. It should be in the HubController later. This user control should just be about logging in and working with an existing session from other components.